SYCL/Plugin/level_zero_batch_test_copy_with_compute.cpp

```diff
@@ -0,0 +1,413 @@
+// REQUIRES: gpu, level_zero
+
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple  %s -o %t.out
+
+// Set batching to 4 explicitly
+// RUN: env SYCL_PI_LEVEL_ZERO_BATCH_SIZE=4 SYCL_PI_TRACE=2 ZE_DEBUG=1 %GPU_RUN_PLACEHOLDER %t.out 2>&1 | FileCheck --check-prefixes=CKALL,CKB4 %s
+
+// Set batching to 1 explicitly
+// RUN: env SYCL_PI_LEVEL_ZERO_BATCH_SIZE=1 SYCL_PI_TRACE=2 ZE_DEBUG=1 %GPU_RUN_PLACEHOLDER %t.out 2>&1 | FileCheck --check-prefixes=CKALL,CKB1 %s
+
+// Set batching to 3 explicitly
+// RUN: env SYCL_PI_LEVEL_ZERO_BATCH_SIZE=3 SYCL_PI_TRACE=2 ZE_DEBUG=1 %GPU_RUN_PLACEHOLDER %t.out 2>&1 | FileCheck --check-prefixes=CKALL,CKB3 %s
+
+// Set batching to 5 explicitly
+// RUN: env SYCL_PI_LEVEL_ZERO_BATCH_SIZE=5 SYCL_PI_TRACE=2 ZE_DEBUG=1 %GPU_RUN_PLACEHOLDER %t.out 2>&1 | FileCheck --check-prefixes=CKALL,CKB5 %s
+
+// Set batching to 7 explicitly
+// RUN: env SYCL_PI_LEVEL_ZERO_BATCH_SIZE=7 SYCL_PI_TRACE=2 ZE_DEBUG=1 %GPU_RUN_PLACEHOLDER %t.out 2>&1 | FileCheck --check-prefixes=CKALL,CKB7 %s
+
+// Set batching to 8 explicitly
+// RUN: env SYCL_PI_LEVEL_ZERO_BATCH_SIZE=8 SYCL_PI_TRACE=2 ZE_DEBUG=1 %GPU_RUN_PLACEHOLDER %t.out 2>&1 | FileCheck --check-prefixes=CKALL,CKB8 %s
+
+// Set batching to 9 explicitly
+// RUN: env SYCL_PI_LEVEL_ZERO_BATCH_SIZE=9 SYCL_PI_TRACE=2 ZE_DEBUG=1 %GPU_RUN_PLACEHOLDER %t.out 2>&1 | FileCheck --check-prefixes=CKALL,CKB9 %s
+
+// level_zero_batch_test_copy_with_compute.cpp
+//
+// This tests the level zero plugin's kernel batching code.  The default
+// batching is 4, and exact batch size can be controlled with environment
+// variable SYCL_PI_LEVEL_ZEOR+BATCH_SIZE=N.
+// This test enqueues 8 kernels and then does a wait. And it does this 3 times.
+// Expected output is that for batching =1 you will see zeCommandListClose,
+// and zeCommandQueueExecuteCommandLists after every piEnqueueKernelLaunch.
+// For batching=3 you will see that after 3rd and 6th enqueues, and then after
+// piEventsWait. For 5, after 5th piEnqueue, and then after piEventsWait.  For
+// 4 you will see these after 4th and 8th Enqueue, and for 8, only after the
+// 8th enqueue.  And lastly for 9, you will see the Close and Execute calls
+// only after the piEventsWait.
+// Since the test does this 3 times, this pattern will repeat 2 more times,
+// and then the test will print Test Passed 8 times, once for each kernel
+// validation check.
+// Pattern starts first set of kernel executions.
```

Comment:
Can you explain how these patterns "catch" copy batching, please? Maybe you should use new zePrint that you've added in https://github.com/intel/llvm/pull/5155?

Reply:
@asudarsa : Maybe you meant that the memory copies you inserted do not interact with compute kernels batching. This is good but I think we should additionally test that the copies are batched too.

Reply:
I suggest to merge this, and improve later.